app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDatabase.java

```diff
@@ -56,6 +56,9 @@ public Task<String> add(String collectionPath, Object data) {
 
     @Override
     public Task<CollectionSnapshot> fetchAll(String collectionPath) {
-        return null;
+        return database.collection(collectionPath).get().onSuccessTask(
+                (collectionSnapshot ->
+                Tasks.forResult(new FirebaseCollectionSnapshot(collectionSnapshot))
+        ));
```

Comment:
Another alternative could have been to use the continueWith function of a Task, which takes as a parameter a continuation object that will implment a "then" method, that takes the task result as a paramter, and returns an object that will be the return type of the new task (sort of a transform). I think it's better that creating a new task each time manually
https://developers.google.com/android/reference/com/google/android/gms/tasks/Continuation